tuist/tuist
# Fix tests crashing when accessing SPM dependency resource bundle

Resolves <https://github.com/tuist/tuist/issues/6885>

### Short description 📝

This fixes the issue of tests crashing when attempting to access a resource that is in an SPM dependency's resource bundle.

The main issue is that when testing a module that depends on another module which depends on an SPM-imported dependency with a resource, none of the URLs used in the lookup of that SPM module's bundle work.
`Bundle.main` appears to refer to the Xcode test agent, and the bundles for the SPM dependency are not copied into the framework, so `Bundle(for: BundleFinder.self).resourceURL` also won't find it. This leads to a `fatalError` because no bundle can be found.

I'm presuming, based on other PR titles, that it's intentional that the `*.bundle` files are no longer copied into the framework for the module being tested. Therefore, I've solved this by adding a new lookup URL in the `Bundle.module` accessor. Here, I'm using `Bundle(for: BundleFinder.self).resourceURL.appendingPathComponent("..")` to look in the parent directory, which *does* appear to contain the appropriate `*.bundle` files for the imported SPM dependency.

### How to test the changes locally 🧐

I've included a new minimum-replicable example project called `framework_with_spm_bundle` that imports the Stripe SDK, as this is the initial framework I encountered the issue with! More than happy to switch this out for another SPM bundle that replicates the same idea, including creating a smaller example of my own, if you'd prefer?

I've added a new test that runs the tests for `framework_with_spm_bundle`. This previously crashed before I added the changes in this PR, and can be tested by checking out the second commit then checking out the most recent commit to confirm it's fixed.

### Contributor checklist ✅


### Reviewer checklist ✅
